Lib/test/test_asyncio/functional.py

```diff
@@ -0,0 +1,283 @@
+import asyncio
+import asyncio.events
+import contextlib
+import os
+import pprint
+import select
+import socket
+import ssl
+import tempfile
+import threading
+
+
+class FunctionalTestCaseMixin:
```

Comment:
Add docstrings in classes and important methods?

Reply:
That's a private unittest helper class in `Lib/test/test_asyncio/`... I copied it from uvloop and I'm still in the process of refactoring it and adapting it to asyncio/Python needs. So I'll add docstrings later.